geode-core/src/main/java/org/apache/geode/distributed/internal/InternalLocator.java

```diff
@@ -1041,7 +1041,10 @@ private void restartWithDS(InternalDistributedSystem newSystem, InternalCache ne
         this.productUseLog.reopen();
       }
       this.productUseLog.monitorUse(newSystem);
-      startSharedConfigurationService();
+      if (isSharedConfigurationEnabled()) {
+        this.sharedConfig = new ClusterConfigurationService(newCache);
+        startSharedConfigurationService();
```

Comment:
there is some logic in startSharedConfigurationService that's already checking isSharedConfigurationEnabled(), maybe we should change the name of this method to be startSharedConfigWhenNeeded, and rework the logic in there to get the new cache. Anyway, just a suggestion to get  cleaner code, but not a must change.